### ISSUE TITLE
Insufficient handling of JNDI based database datastore 

### ISSUE THREAD

**Issue by r0bb3n** [geosolutions-it/geoserver-manager#143]
I use a  `Oracle NG (JNDI)` datastore which does not have the config parameter `database`. Thus the retrieved `RESTDataStore` does not contain this connection parameter which leads to an `IllegalArgumentException` because of a _not null_ check in the according constructor.

```
    public GSOracleNGDatastoreEncoder(RESTDataStore store) {
    	super(store);

    	// Check mandatory parameter validity
		ensureValidDatabase(store.getConnectionParameters().get("database")); // <- IllegalArgumentException !
    }
```

I think a separation of all JNDI and non-JNDI database datastores would be a better approach due to their differences (more or less available setters for optional/required parameters).

**Comment by mbarto** [comment]
I agree. If you already worked on it and would like to prepare a pull request, it would be much appreciated.
